src/compiler/binder.ts

```diff
@@ -2456,13 +2456,23 @@ namespace ts {
                 }
                 else {
                     for (const j of node.jsDoc!) {
-                        setParent(j, node);
-                        setParentRecursive(j, /*incremental*/ false);
+                        setParentsAndBindAnonymousJSDocTypes(j);
                     }
                 }
             }
         }
 
+        function setParentsAndBindAnonymousJSDocTypes(node: Node) {
+            setParent(node, parent);
+            const saveParent = parent;
+            parent = node;
+            if (node.kind === SyntaxKind.TypeLiteral) {

```

Comment:
won't this also bind type literals inside a `@typedef`? Maybe it won't matter because there's some required intermediate binding needed for the checker to check the type.